cmd/kubeadm/app/apis/kubeadm/types.go

```diff
@@ -92,7 +92,23 @@ type Etcd struct {
 	DataDir   string
 	ExtraArgs map[string]string
 	// Image specifies which container image to use for running etcd. If empty, automatically populated by kubeadm using the image repository and default etcd version
-	Image string
+	Image      string
+	SelfHosted *SelfHostedEtcd
+}
+
+// SelfHostedEtcd describes options required to configure self-hosted etcd
+type SelfHostedEtcd struct {
+	// CertificatesDir represents the directory where all etcd TLS assets are stored. By default this is
+	// a dir names "etcd" in the mains CertificatesDir value.
+	CertificatesDir string
+	// The service IP of the etcd cluster
+	ClusterIP string
+	// The number of members that will exist in the etcd cluster.
+	ClusterSize int
+	// The version of etcd running in the cluster.
+	EtcdVersion string
+	// The version of the etcd-operator to use.
```

Comment:
nit here and elsewhere, most often Godoc requires `// Foo foos the bar` instead of `// foos the bar`